pkg/reconciler/security/istio/eventpolicybinding/eventpolicybinding.go

```diff
@@ -0,0 +1,146 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventpolicybinding
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/tracker"
+
+	"github.com/google/knative-gcp/pkg/apis/security/v1alpha1"
+	bindingreconciler "github.com/google/knative-gcp/pkg/client/injection/reconciler/security/v1alpha1/eventpolicybinding"
+	securitylisters "github.com/google/knative-gcp/pkg/client/listers/security/v1alpha1"
+	"github.com/google/knative-gcp/pkg/reconciler"
+	"github.com/google/knative-gcp/pkg/reconciler/security/istio/eventpolicybinding/resources"
+)
+
+// Reconciler reconciles the EventPolicyBinding.
+type Reconciler struct {
+	*reconciler.Base
+
+	eventPolicyLister       securitylisters.EventPolicyLister
+	httpPolicyLister        securitylisters.HTTPPolicyLister
+	httpPolicyBindingLister securitylisters.HTTPPolicyBindingLister
+
+	policyTracker tracker.Interface
+}
+
+var _ bindingreconciler.Interface = (*Reconciler)(nil)
+
+// ReconcileKind reconciles the EventPolicyBinding.
+func (r *Reconciler) ReconcileKind(ctx context.Context, b *v1alpha1.EventPolicyBinding) pkgreconciler.Event {
+	logging.FromContext(ctx).Debug("Reconciling", zap.Any("EventPolicyBinding", b))
+	b.Status.InitializeConditions()
+	b.Status.ObservedGeneration = b.Generation
+
+	p, err := r.eventPolicyLister.EventPolicies(b.Spec.Policy.Namespace).Get(b.Spec.Policy.Name)
+	if err != nil {
+		logging.FromContext(ctx).Error("Problem getting EventPolicy", zap.Any("EventPolicy", b.Spec.Policy), zap.Error(err))
+		b.Status.MarkBindingFailure("GetPolicyFailure", "%v", err)
+		return fmt.Errorf("failed to get EventPolicy: %w", err)
+	}
+	// Track referenced policy.
+	r.policyTracker.TrackReference(tracker.Reference{
+		APIVersion: eventPolicyGVK.GroupVersion().String(),
+		Kind:       eventPolicyGVK.Kind,
+		Namespace:  b.Spec.Policy.Namespace,
+		Name:       b.Spec.Policy.Name,
+	}, b)
+
+	hp, err := r.reconcileHTTPPolicy(ctx, b, p)
+	if err != nil {
+		logging.FromContext(ctx).Error("Problem reconciling HTTPPolicy", zap.Error(err))
+		b.Status.MarkBindingFailure("HTTPPolicyReconcileFailure", "%v", err)
+		return err
+	}
+
+	hpb, err := r.reconcileHTTPPolicyBinding(ctx, b, hp)
+	if err != nil {
+		logging.FromContext(ctx).Error("Problem reconciling HTTPPolicyBinding", zap.Error(err))
+		b.Status.MarkBindingFailure("HTTPPolicyBindingReconcileFailure", "%v", err)
+		return err
+	}
+
+	// Copy the HTTP policy binding status.
+	b.Status = *hpb.Status.DeepCopy()
```

Comment:
do you need this DeepCopy here?

Reply:
I think it might be dangerous to deep copy. So now I changed to explicitly set status.